scripts/sommelier

```diff
@@ -67,6 +67,21 @@ init_wine() {
       mv "$WINEPREFIX" "${WINEPREFIX}.bak-$(date +%s)"
   fi
 
+  if [[ -n "${MONO_URL:-}" ]]; then
```

Comment:
I would prefer sommelier-core provides this URL if possible. So users can just set a flag "I want mono to be installed".

Reply:
we could add it this would require to have each mono version url and check it with flag envar having mono version on it still will require updating in case of future versions.

Reply:
it should be a build time application of the sommelier core snaps and be irrelevant to end consumer snaps.

Reply:
The required mono version depends on the Wine version. We don't know at build time what Wine version we'll be using since this depends on the content snap connected at runtime. Therefore, I don't think we can add `wine-mono` at build time.

Ideally, we automatically check at runtime what Mono version is required and download the right version. So we have a flag `SOMMELIER_INSTALL_MONO=1`. If this flag is set, we check which version of Mono is required by Wine and download that version.

However, I'm currently not finding a good way to extract the required mono version, though. Our options:

* A table [in this page](https://wiki.winehq.org/Mono) maps wine version to mono version, but I'd rather not parse HTML.
* It's in [appwiz.dll](https://github.com/wine-mirror/wine/blob/e909986e6ea5ecd49b2b847f321ad89b2ae4f6f1/dlls/appwiz.cpl/addons.c#L60) but I don't know of a reliable way to extract that version from the dll itself.
* Getting the wine version is easy to do from the CLI so we could add a table mapping Wine version to Mono version in sommelier-core. But then we need to update this table each time we release a new platform snap.

Reply:
* Checking mono version from file `appwiz.dll.so` in `wine 3.x` yeilds the result but on newer version doesn't have `appwiz.dll.so` due to PE conversion but on non PE build same file doesn't have string. So its not usefull for us following this method
*  we could add all mono versions in script then match mono versions with wine and assign.
* we can check wine version from file `libwine.so.1.0` from content snaps without using `wine --version`

Reply:
> * we could add all mono versions in script then match mono versions with wine and assign.
> * we can check wine version from file libwine.so.1.0 from content snaps without using wine --version

Ok, then I think this is the best solution currently. I think you can implement this, unless @diddledani has objections. Just curious, what is the advantage of using `libwine.so.1.0` instead of `wine --version`?

Reply:
we grep version string from lib file on the otherhand we are exec wine inorder to check version.